.github/workflows/_github-release.yml

```diff
@@ -0,0 +1,166 @@
+# **what?**
+# If no release already exists for this commit and version, create the tag and release it to GitHub.
+# If a release already exists for this commit, skip creating the release but finish with a success.
+# If a release exists for this commit under a different tag, fail.
+#
+# Inputs:
+#  sha: the commit to attache to this release
+#  version_number: version number for the release (ex: 1.2.3rc2)
+#  file_uploads: Newline-delimited globs of paths to assets to upload for release
+#   
+# **why?**
+# Reusable and consistent GitHUb release process.
+#
+# **when?**
+# Call after the build process.
+#
+# This workflow expects the artifacts to already be built and living in the artifact store of the workflow.
+#
+
+name: GitHub Release
+
+on:
+  workflow_call:
+    inputs:
+      sha:
```

Comment:
Same thought of passing tag here with it already being created before this step

Reply:
Are we manually creating the tag now?  The action `softprops/action-gh-release@v1` will create the tag for us.  This is designed with the assumption that's how the flow should work.